### PR TITLE
Add VisitNote Model

### PR DESCRIPTION
**Pull request by trevorskaggs**

### CODE REVIEW

**Comment by trevorskaggs** [comment]
@AlexMountain ready for review! Have to manually go to the "Edit VisitNote" page.  URL form is` /evac/assignment/note/<pk>`

File: frontend/src/evac/EvacForms.js [hunk 1]
```diff
@@ -109,4 +113,103 @@ export const EvacTeamMemberForm = () => {
...
+                </Row>
+                <Row>
+                  <Col xs={{size: 2}}>
+                    <TextInput
+                      as="textarea"
+                      label="Notes*"
+                      name="notes"
+                      id="notes"
+                      xs="4"
+                      rows={3}
```
Comment (AlexMountain) [inline frontend/src/evac/EvacForms.js:201]: Thoughts on making this bigger?  e.g. `xs="9"` and `rows={5}`?

File: frontend/src/evac/EvacForms.js [hunk 2]
```diff
@@ -109,4 +113,103 @@ export const EvacTeamMemberForm = () => {
...
+                  <Col xs={{size: 2}}>
+                  <DateTimePicker
+                    label="Date Completed"
+                    name="date_completed"
+                    id="date_completed"
+                    xs="4"
+                    onChange={(date, dateStr) => {
+                      form.setFieldValue("date_completed", dateStr)
+                    }}
+                    value={form.values.date_completed||null}
```
Comment (AlexMountain) [inline frontend/src/evac/EvacForms.js:189]: Do we want to even allow this to be clearable?  If we want to force it to stay populated, I added the ability to hide the clear button in my current ticket:

here: https://github.com/trevorskaggs/shelterly/pull/120/files#diff-a2f467e702bf781a7d25b394512b80e51204d6015752b5c74629c7df17248af8R255
here: https://github.com/trevorskaggs/shelterly/pull/120/files#diff-3675db6a5515c78e05033de36f7bda92123119edfd29ea79ecbdf227108d0551R16
and here: https://github.com/trevorskaggs/shelterly/pull/120/files#diff-3675db6a5515c78e05033de36f7bda92123119edfd29ea79ecbdf227108d0551R43

File: frontend/src/evac/EvacForms.js [hunk 3]
```diff
@@ -109,4 +113,103 @@ export const EvacTeamMemberForm = () => {
...
+                      name="notes"
+                      id="notes"
+                      xs="4"
+                      rows={3}
+                    />
+                  </Col>
+                </Row>
+              </FormGroup>
+              <Button type="button" className="btn btn-success mr-1" onClick={() => form.submitForm()}>Save</Button>
+              <Link className="btn btn-secondary" href="/evac">Cancel</Link>
```
Comment (AlexMountain) [inline frontend/src/evac/EvacForms.js:207]: Cancel button effectively redundant with the back button?

File: evac/views.py [hunk 4]
```diff
@@ -29,3 +29,10 @@ def perform_create(self, serializer):
             action.send(self.request.user, verb='created evacuation assignment', target=evac_assignment)
             for service_request in service_requests:
                 action.send(self.request.user, verb='assigned service request', target=service_request)
+
+class VisitNoteViewSet(viewsets.ModelViewSet):
+
+    queryset = VisitNote.objects.all()
+    filter_backends = (filters.SearchFilter,)
```
Comment (AlexMountain) [inline evac/views.py:36]: Do we need the SearchFilter here?

File: frontend/src/evac/EvacForms.js [hunk 5]
```diff
@@ -109,4 +113,103 @@ export const EvacTeamMemberForm = () => {
...
+};
+
+export const VisitNoteForm = ({id}) => {
+
+    const [data, setData] = useState({
+      date_completed: '',
+      owner_contacted: '',
+      notes: '',
+      service_request: '',
+      evac_assignment: '',
```
Comment (AlexMountain) [inline frontend/src/evac/EvacForms.js:125]: Default should probably be null for FKs and date_completed. owner_contacted should probably be a bool?

File: evac/models.py [hunk 6]
```diff
@@ -23,3 +23,12 @@ class EvacAssignment(models.Model):
...
     end_time = models.DateTimeField(blank=True, null=True)
+
+
+class VisitNote(models.Model):
+
+    date_completed = models.DateTimeField()
+    evac_assignment = models.ForeignKey(EvacAssignment, on_delete=models.CASCADE)
+    service_request = models.ForeignKey(ServiceRequest, on_delete=models.CASCADE)
+    owner_contacted = models.BooleanField(default=False)
+    notes = models.CharField(max_length=200, blank=True)
```
Comment (AlexMountain) [inline evac/models.py:34]: Just want to make sure we think 200 will be enough?  why blank=True?

**Review by AlexMountain — COMMENTED** [review]
This functionality will be a great add!  Just some minor suggestions.

File: evac/models.py (hunk 6 above)
Comment (trevorskaggs) [reply evac/models.py:34]: Can up the amount, but we dont need a note per se, but we do need a date_completed to be tied up with owner_contacted for a given visit.  There can be no notes though.

Comment (AlexMountain) [reply evac/models.py:34]: Roger.  In that case we might want to remove the `*` from the Notes field in the UI.

**Comment by trevorskaggs** [comment]
Re-Ready for review!

**Review by AlexMountain — APPROVED** [review]
Solid!  Let's get this bad boy merged in!  🎶

**Pull request merged**